src/components/modal/modal.ts

```diff
@@ -0,0 +1,196 @@
+import {
+  bind,
+  forwardRef,
+  Component,
+  ComponentRef,
+  Directive,
+  DynamicComponentLoader,
+  ElementRef,
+  Host,
+  Injectable,
+  ResolvedBinding,
+  SkipSelf,
+  Injector,
+  View,
+  ViewEncapsulation
+} from 'angular2/core';
+
+import {ObservableWrapper, Promise, PromiseWrapper} from 'angular2/src/core/facade/async';
+import {isPresent, Type} from 'angular2/src/core/facade/lang';
+import {DOM} from 'angular2/src/core/dom/dom_adapter';
+import {MouseEvent, KeyboardEvent} from 'angular2/src/core/facade/browser';
+import {KeyCodes} from 'angular2_material/src/core/key_codes';
+
+@Injectable()
+export class UibModal {
+  componentLoader: DynamicComponentLoader;
+
+  constructor(loader: DynamicComponentLoader) {
+    this.componentLoader = loader;
```

Comment:
Common convention I am seeing (but subjective I guess) is to put `private` on the constructor so we don't need line 29 and 26.